libraries/ESP8266WebServer/examples/Graph/Graph.ino

```diff
@@ -0,0 +1,318 @@
+/*
+  Graph - A web-based Graph display of ESP8266 data
+
+  This file is part of the ESP8266WebServer library for Arduino environment.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+  See readme.md for more information.
+*/
+
+////////////////////////////////
+
+// Select the FileSystem by uncommenting one of the lines below
+
+//#define USE_SPIFFS
+#define USE_LITTLEFS
+//#define USE_SDFS
+
+////////////////////////////////
+
+#include <ESP8266WiFi.h>
+#include <WiFiClient.h>
+#include <ESP8266WebServer.h>
+#include <ESP8266mDNS.h>
+#include <SPI.h>
+
+#if defined USE_SPIFFS
+#include <FS.h>
+FS* fileSystem = &SPIFFS;
+SPIFFSConfig fileSystemConfig = SPIFFSConfig();
+#elif defined USE_LITTLEFS
+#include <LittleFS.h>
+FS* fileSystem = &LittleFS;
+LittleFSConfig fileSystemConfig = LittleFSConfig();
+#elif defined USE_SDFS
+#include <SDFS.h>
+FS* fileSystem = &SDFS;
+SDFSConfig fileSystemConfig = SDFSConfig();
+// fileSystemConfig.setCSPin(chipSelectPin);
+#else
+#error Please select a filesystem first by uncommenting one of the "#define USE_xxx" lines at the beginning of the sketch.
+#endif
+
+
+#define DBG_OUTPUT_PORT Serial
+
+#ifndef STASSID
+#define STASSID "your-ssid"
+#define STAPSK  "your-password"
+#endif
+
+// Indicate which digital I/Os should be displayed on the chart.
+// From GPIO16 to GPIO0, a '1' means the corresponding GPIO will be shown
+// Note: SD GPIOs are hidden by default:
+#define DEFAULT_GPIO_MASK 0b11111000000111111
+
+unsigned int gpioMask = DEFAULT_GPIO_MASK;
+
```

Comment:
A better way of getting the "safe" GPIO mask programatically using the `isFlashInterface` API
````
mask = 0x1ffff;
for (auto pin=0; pin <= 16; pin++) if (isFlashInterfacePin(pin)) mask &= ~(1<<pin);
````

Reply:
Cool, didn't know that. Will change accordingly.

Reply:
Done. Thanks for the tip.